app/src/main/java/com/fstyle/structure_android/data/repository/UserRepository.java

```diff
@@ -37,196 +22,101 @@
+            @NonNull UserRemoteDataSource userRemoteDataSource) {
+        mUserLocalDataSource = checkNotNull(userLocalDataSource);
+        mUserRemoteDataSource = checkNotNull(userRemoteDataSource);
+    }
+
+    /**
+     * Returns the single instance of this class, creating it if necessary.
+     *
+     * @param remoteDataSource the backend data source
+     * @param localDataSource the device storage data source
+     * @return the {@link UserRepository} instance
+     */
+    public static synchronized UserRepository getInstance(
+            @NonNull UserLocalDataSource localDataSource,
+            @NonNull UserRemoteDataSource remoteDataSource) {
+        if (INSTANCE == null) {
+            INSTANCE = new UserRepository(checkNotNull(localDataSource),
+                    checkNotNull(remoteDataSource));
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Used to force {@link #getInstance(UserLocalDataSource, UserRemoteDataSource)} to create a new
+     * instance
+     * next time it's called.
+     */
+    public static void destroyInstance() {
+        INSTANCE = null;
     }
 
     @Override
-    public Single<List<User>> searchUsers(String keyWord, String limit) {
-        return mNameApi.searchGithubUsers(keyWord, limit)
-                .flatMap(new Function<SearchUserResponse, SingleSource<? extends List<User>>>() {
-                    @Override
-                    public SingleSource<? extends List<User>> apply(
-                            SearchUserResponse searchUserResponse) throws Exception {
-                        return Single.just(searchUserResponse.getItems());
-                    }
-                });
+    public Completable insertListUser(List<User> users) {
+        return mUserLocalDataSource.insertListUser(Preconditions.checkNotNull(users));
     }
 
     @Override
-    public Completable insertUser(@NonNull final User user) {
-        mDatabase.beginTransaction();
-        return Completable.create(new CompletableOnSubscribe() {
-            @Override
-            public void subscribe(CompletableEmitter emitter) throws Exception {
-                try {
-                    ContentValues values = new ContentValues();
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
-                            user.getSubscriptionsUrl());
-                    mDatabase.insert(UserDbHelper.UserEntry.TABLE_NAME, null, values);
-                    mDatabase.close();
-                    emitter.onComplete();
-                } catch (RuntimeException e) {
-                    emitter.onError(e);
-                }
-            }
-        });
+    public Completable insertUser(@NonNull User user) {
+        return mUserLocalDataSource.insertUser(Preconditions.checkNotNull(user));
```

Comment:
WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Preconditions.checkNotNull' due to existing static import 'com.google.common.base.Preconditions.checkNotNull'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName